frontend-ssr/redux/exercise-submissions/exercise-submissions.reducer.ts

```diff
@@ -0,0 +1,54 @@
+import { SubmissionState } from './types';
+import { EXERCISE_SUBMISSIONS } from './exercise-submissions.actions';
+
+const initialState: SubmissionState = {
+  submissions: [],
+};
+export const submissionReducer = (state = initialState, action: { type: string; payload: any;}) => {
+  switch (action.type) {
+    case EXERCISE_SUBMISSIONS.ADD_SUBMISSION: {
+      const { index, submission } = action.payload;
+      const newArray = [...state.submissions];
+      newArray.splice(index, 0, submission);
+      if (Array.isArray(submission)) {
+        submission.map((item) => ({
```

Comment:
Did you test this. Doesn't seem like the result of this map is being put anywhere. Looks a little wrong as well

Also, I think we can simply to something like:

```
[...state.submissions.slice(0, index),  ...submission, ...state.submissions.slice(index)]
```